pkg/storaged/btrfs/device.jsx

```diff
@@ -0,0 +1,89 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import client from "../client";
+
+import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.js";
+import { CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+
+import { StorageCard, StorageDescription, new_card, register_crossref } from "../pages.jsx";
+import { StorageUsageBar } from "../storage-controls.jsx";
+import { btrfs_device_usage } from "./utils.jsx";
+
+const _ = cockpit.gettext;
+
+export function make_btrfs_device_card(next, backing_block, content_block) {
+    const block_btrfs = client.blocks_fsys_btrfs[content_block.path];
+    const label = block_btrfs && block_btrfs.data.label;
+    const uuid = block_btrfs && block_btrfs.data.uuid;
+    const use = btrfs_device_usage(client, uuid, block_btrfs.path);
+
+    const btrfs_card = new_card({
+        title: _("btrfs device"),
+        location: label || uuid,
+        next,
+        component: BtrfsDeviceCard,
+        props: { backing_block, content_block },
+    });
+
+    if (uuid) {
+        register_crossref({
+            key: uuid,
+            card: btrfs_card,
+            actions: [
+            ],
+            size: <StorageUsageBar stats={use} short />,
+        });
+    }
+
+    return btrfs_card;
+}
+
+export const BtrfsDeviceCard = ({ card, backing_block, content_block }) => {
+    const block_btrfs = client.blocks_fsys_btrfs[content_block.path];
+    const uuid = block_btrfs && block_btrfs.data.uuid;
+    const label = block_btrfs && block_btrfs.data.label;
+    const use = btrfs_device_usage(client, uuid, block_btrfs.path);
+
+    return (
+        <StorageCard card={card}>
+            <CardBody>
+                <DescriptionList className="pf-m-horizontal-on-sm">
+                    <StorageDescription title={_("btrfs volume")}>
+                        {uuid
+                            ? <Button variant="link" isInline role="link"
+                                   onClick={() => cockpit.location.go(["btrfs-volume", uuid])}>
+                                {label || uuid}
+                            </Button>
+                            : "-"
+                        }
+                    </StorageDescription>
+                    <StorageDescription title={_("UUID")} value={content_block.IdUUID} />
+                    { block_btrfs &&
+                    <StorageDescription title={_("Usage")}>
+                        <StorageUsageBar key="s" stats={use} />
+                    </StorageDescription>
+                    }
+                </DescriptionList>
+            </CardBody>
+        </StorageCard>);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110900-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)